stdlib/public/core/CString.swift

```diff
@@ -15,19 +15,50 @@
 import SwiftShims
 
 extension String {
+  /// Create a new `String` by copying the nul-terminated UTF-8 data
+  /// referenced by a `cString`.
+  ///
+  /// If `cString` contains ill-formed UTF-8 code unit sequences, replaces them
+  /// with replacement characters (U+FFFD).
+  ///
+  /// - Requires: `cString != nil`
+  public init(cString: UnsafePointer<CChar>) {
+    precondition(cString != nil, "cString must not be nil")
+    let len = Int(_swift_stdlib_strlen(cString))
+    let buf = UnsafeBufferPointer<UTF8.CodeUnit>(start: UnsafePointer(cString),
+      count: len)
+    self.init(codeUnits: buf, as: UTF8.self)
+  }
+
+  /// Create a new `String` by copying the nul-terminated UTF-8 data
+  /// referenced by a `cString`.
+  ///
+  /// Does not try to repair ill-formed UTF-8 code unit sequences, fails if any
+  /// such sequences are found.
+  ///
+  /// - Requires: `cString != nil`
+  public init?(validatingCString cString: UnsafePointer<CChar>) {
+    precondition(cString != nil, "cString must not be nil")
+    let len = Int(_swift_stdlib_strlen(cString))
+    let buf = UnsafeBufferPointer<UTF8.CodeUnit>(start: UnsafePointer(cString),
+      count: len)
+    self.init(validatingCodeUnits: buf, as: UTF8.self)
+  }
+}
+
+extension String {
   /// Creates a new `String` by copying the nul-terminated UTF-8 data
   /// referenced by a `CString`.
   ///
   /// Returns `nil` if the `CString` is `NULL` or if it contains ill-formed
   /// UTF-8 code unit sequences.
   @warn_unused_result
+  @available(*, deprecated, message="Use String(validatingCString:) initializer")
```

Comment:
I don't know if there is any limit to the length of the message, so... maybe something along the lines of "Use String(cString:) or String(validatingCString:) with nullness check instead".
One unrelated note: in our design, argument for a failing initializer is called `validatingUtf8`, as `validatingCString` does not really reflect what gets validated (which is the UTF8 code units). What do you think?